hosts/hetzner-ephemeral-beta/default.nix

```diff
@@ -0,0 +1,90 @@
+{ pkgs, config, inputs, lib, ... }:
+{
+  # User options
+  users = {
+    juuso.authorizedKeys = [ "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBNMKgTTpGSvPG4p8pRUWg1kqnP9zPKybTHQ0+Q/noY5+M6uOxkLy7FqUIEFUT9ZS/fflLlC/AlJsFBU212UzobA= ssh@secretive.sandbox.local" ];
+    kari.authorizedKeys = [ "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKEdpdbTOz0h9tVvkn13k1e8X7MnctH3zHRFmYWTbz9T kari@torque" ];
+  };
+
+  # Users group
+  users.groups.users = { };
+
+  # Localization
+  networking.hostName = "hetzner-ephemeral-beta";
+  time.timeZone = "Europe/Helsinki";
+
+  # Use stable kernel
+  boot.kernelPackages = pkgs.linuxPackagesFor (pkgs.linux);
+
+  # Apply kernel personality patch from Ubuntu and configure armv7l support
+  boot.kernelPatches = [
+    rec {
+      name = "compat_uts_machine";
+      patch = pkgs.fetchpatch {
+        inherit name;
+        url = "https://git.launchpad.net/~ubuntu-kernel/ubuntu/+source/linux/+git/jammy/patch/?id=c1da50fa6eddad313360249cadcd4905ac9f82ea";
+        sha256 = "sha256-mpq4YLhobWGs+TRKjIjoe5uDiYLVlimqWUCBGFH/zzU=";
+      };
+    }
+  ];
+  boot.kernelParams = [
+    "compat_uts_machine=armv7l"
+  ];
+  nix.extraOptions = "extra-platforms = armv7l-linux";
+
+  # Hercules CI
+  services.hercules-ci-agent = {
+    enable = true;
+    settings.baseDirectory = "/var/mnt/.config/hercules-ci-agent";
+  };
+
+  systemd.mounts = [
+    {
+      enable = true;
+
+      what = "/dev/disk/by-label/nix";
+      where = "/var/mnt/.config";
+      type = "btrfs";
+      options = "subvolid=257";
+
+      before = [ "hercules-ci-agent.service" ];
+      wantedBy = [ "multi-user.target" ];
+    }
+  ];
+
+  systemd.services.nix-remount = {
+    path = [ "/run/wrappers" ];
+    enable = true;
+
+    serviceConfig = {
+      Type = "oneshot";
+    };
+
+    # if a new device, the new .rw-store has to be mounted
+    preStart = ''
+      /run/wrappers/bin/mount /dev/disk/by-label/nix -o subvolid=256 /nix/.rw-store
+    '';
```

Comment:
What should be appended for preStart for reproducibility is:
`mkdir /nix/.rw-store/work`
`mkdir /nix/.rw-store/store`

Thie ensures that the folders mounted by the overlay do exist. This is not the case when the server is new.

This change should also be done for the alpha server.